packages/shared/src/components/post/RelatedPostsWidget.tsx

```diff
@@ -0,0 +1,188 @@
+import React, { ReactElement, useCallback, useState } from 'react';
+import classNames from 'classnames';
+import { useInfiniteQuery, useQueryClient } from '@tanstack/react-query';
+import { widgetClasses } from '../widgets/common';
+import { InfinitePaginationActions } from '../pagination';
+import { ElementPlaceholder } from '../ElementPlaceholder';
+import { RequestKey, StaleTime, generateQueryKey } from '../../lib/query';
+import { useRequestProtocol } from '../../hooks/useRequestProtocol';
+import { graphqlUrl } from '../../lib/config';
+import {
+  Post,
+  PostData,
+  PostRelationType,
+  RELATED_POSTS_PER_PAGE_DEFAULT,
+  RELATED_POSTS_QUERY,
+  RelatedPost,
+} from '../../graphql/posts';
+import { Connection } from '../../graphql/common';
+import { SourceAvatar } from '../profile/source';
+import PostMetadata from '../cards/PostMetadata';
+import { CardLink } from '../cards/Card';
+import { getPostByIdKey } from '../../hooks/usePostById';
+
+export type RelatedPostsWidgetProps = {
+  className?: string;
+  post: Post;
+  perPage?: number;
+  relationType: PostRelationType;
+};
+
+export const RelatedPostsWidget = ({
+  className,
+  post,
+  perPage = RELATED_POSTS_PER_PAGE_DEFAULT,
+  relationType,
+}: RelatedPostsWidgetProps): ReactElement => {
+  const { requestMethod } = useRequestProtocol();
+  const [page, setPage] = useState(0);
+  const queryClient = useQueryClient();
+
+  const {
```

Comment:
This file feels a bit clunky, not blocking but perhaps the query and logic could live in a hook?

Reply:
For sure, moved to hook.